test/src/unittests/tonal/test_pitchcontours.py

```diff
@@ -0,0 +1,138 @@
+        emptyPeakSaliences = []
+        theHopSize= 2 * defaultHopSize
+        _,  _, _, duration = PitchContours(hopSize=theHopSize)(emptyPeakBins, emptyPeakSaliences)       
+        calculatedDuration = (2 * theHopSize)/defaultSampleRate
+        self.assertAlmostEqual(duration, calculatedDuration, 8)
+        
+        theHopSize= 4 * defaultHopSize        
+        _,  _, _, duration = PitchContours(hopSize=theHopSize)(emptyPeakBins, emptyPeakSaliences)               
+        calculatedDuration = (2 * theHopSize)/defaultSampleRate        
+        self.assertAlmostEqual(duration, calculatedDuration, 8)
+
+        theHopSize= 8 * defaultHopSize        
+        _,  _, _, duration = PitchContours(hopSize=theHopSize)(emptyPeakBins, emptyPeakSaliences)               
+        calculatedDuration = (2 * theHopSize)/defaultSampleRate        
+        self.assertAlmostEqual(duration, calculatedDuration, 8)                        
+        
+    def testEmpty(self):
+        emptyPeakBins = []
+        emptyPeakSaliences = []
+        bins, saliences, startTimes, duration = PitchContours()(emptyPeakBins, emptyPeakSaliences)       
+        self.assertEqualVector(bins, [])
+        self.assertEqualVector(saliences, [])
+        self.assertEqualVector(startTimes, [])
+        # Previous tests showed small duration of 0.0058 seconds for zero or empty inputs.        
+        calculatedDuration = (2*defaultHopSize)/defaultSampleRate
+        self.assertAlmostEqual(duration, calculatedDuration, 8)
+
+    def testEmptyColumns(self):
+        emptyPeakBins = [[],[]]
+        emptyPeakSaliences = [[],[]]
+        theHopSize= 2*defaultHopSize
+        bins, saliences, startTimes, duration = PitchContours(hopSize=theHopSize)(emptyPeakBins, emptyPeakSaliences)       
+        self.assertEqualVector(bins, [])
+        self.assertEqualVector(saliences, [])
+        self.assertEqualVector(startTimes, [])
+        # Previous tests showed small duration of 0.0058 seconds for zero or empty inputs.        
+        calculatedDuration = (2*theHopSize)/defaultSampleRate
+        self.assertAlmostEqual(duration, calculatedDuration, 8)
+
+    def testZeros(self):
+        theHopSize = 4*defaultHopSize
+        bins, saliences, startTimes, duration = PitchContours(hopSize=theHopSize)(array(zeros([2,256])), array(zeros([2,256])))      
+        self.assertEqualVector(bins, [])
+        self.assertEqualVector(saliences, [])
+        self.assertEqualVector(startTimes, [])
+        # Previous tests showed small duration of 0.0058 seconds for zero or empty inputs.
+        calculatedDuration = (2*theHopSize)/defaultSampleRate            
+        self.assertAlmostEqual(duration, calculatedDuration, 8)
+
+    #256 frames x single zero-salience peak at the same cent bin in each frame.
+    def testSingleZeroSaliencePeak(self):
+        peakBins = array(zeros([1, 256]))
+        peakSaliences = array(zeros([1, 256]))
+
+        theHopSize = 4*defaultHopSize
```

Comment:
It is unclear why we need to change the default hop size recommended for pitch contour estimation.

Reply:
duration is calculated as follows in the code:

The test wants to examine different duration outcomes with different hop sizes
duration = _numberFrames * _frameDuration;